doc: Add user journey considerations in current resource and example documentation (#2559)
mongodb/terraform-provider-mongodbatlas: docs/resources/encryption_at_rest_private_endpoint.md

`mongodbatlas_encryption_at_rest_private_endpoint` provides a resource for managing a private endpoint used for encryption at rest with customer-managed keys. This ensures all traffic between Atlas and customer key management systems take place over private network interfaces.

> ~> **IMPORTANT** The Encryption at Rest using Azure Key Vault over Private Endpoints feature is available by request. To request this functionality for your Atlas deployments, contact your Account Manager. 

Reviewer: Can we also add a note in this resource doc that users need to set `azure_key_vault_config.require_private_networking` in EAR resource to use this?

> Additionally, you'll need to set the environment variable `MONGODB_ATLAS_ENABLE_PREVIEW=true` to use this resource. To learn more about existing limitations, see the [Manage Customer Keys with Azure Key Vault Over Private Endpoints](https://www.mongodb.com/docs/atlas/security/azure-kms-over-private-endpoint/#manage-customer-keys-with-azure-key-vault-over-private-endpoints).

Reviewer: can we also include[ this point ](https://docs.google.com/document/d/1HqnXptK4-kkCDVFtwEh3JaW1cQ0iVQIODTy2RTQ8g14/edit#bookmark=id.hjiv1p74yrrn)about migrating from public to private networking?

Author: included a mention for attribute that needs to be defined, and the fact that it can be updated in place

`mongodbatlas_encryption_at_rest_private_endpoint` provides a resource for managing a private endpoint used for encryption at rest with customer-managed keys. This ensures all traffic between Atlas and customer key management systems take place over private network interfaces.

> ~> **IMPORTANT** The Encryption at Rest using Azure Key Vault over Private Endpoints feature is available by request. To request this functionality for your Atlas deployments, contact your Account Manager. 

Reviewer: I think this should also be included in data source doc otherwise I believe users will get error, WDYT?

Author: included 👍

Additionally, you'll need to set the environment variable `MONGODB_ATLAS_ENABLE_PREVIEW=true` to use this resource. To learn more about existing limitations, see the [Manage Customer Keys with Azure Key Vault Over Private Endpoints](https://www.mongodb.com/docs/atlas/security/azure-kms-over-private-endpoint/#manage-customer-keys-with-azure-key-vault-over-private-endpoints).

> -> **NOTE:** As a prerequisite to configuring a private endpoint for Azure Key Vault, the corresponding [`mongodbatlas_encryption_at_rest`](encryption_at_rest) resource has to be adjust by configuring [`azure_key_vault_config.require_private_networking`](encryption_at_rest#require_private_networking) to true. This attribute can be updated in place, ensuring the customer-managed keys configuration is never disabled.

Reviewer: ```suggestion
-> **NOTE:** As a prerequisite to configuring a private endpoint for Azure Key Vault, the corresponding [`mongodbatlas_encryption_at_rest`](encryption_at_rest) resource has to be adjust by configuring [`azure_key_vault_config.require_private_networking`](encryption_at_rest#require_private_networking) to true. This attribute should be updated in place, ensuring the customer-managed keys encryption is never disabled.
```

-> **NOTE:** This resource does not support update operations. To modify values of a private endpoint the existing resource must be deleted and a new one can be created with the modified values.

## Example Usages

> -> **NOTE:** Only Azure Key Vault with Azure Private Link is supported at this time.

Reviewer: Should we also include this in the data source documentation?

Author: no harm in clarifying there as well since cloud_provider is a required attribute for both data sources.
